frontend/src/utils/MeshUtils.ts

```diff
@@ -0,0 +1,17 @@
+import { store } from '../store/ConfigStore';
+import { serverConfig } from '../config';
+
+// isControlPlaneAccessible returns true if
+export const isControlPlaneAccessible = (cluster?: string): boolean => {
+  const ns = store.getState().namespaces;
+
+  return (
+    ns.items !== undefined &&
+    ns.items.some(nsItem => {
+      if (cluster) {
+        return nsItem.name === serverConfig.istioNamespace && nsItem.cluster === cluster;
+      }
+      return nsItem.name === serverConfig.istioNamespace;
+    })
+  );
+};
```

Comment:
This could also just be a redux selector I think instead of a separate function. I'm not sure if it's common to import `store` directly or if that has other side effects.

Reply:
You can use it to access to the redux state outside of the React context. The only thing is that it wouldn't refresh automatically with a state change (But I don't think it is required in this context)

Reply:
Although using a Redux selector is the recommended approach, I think there is no problem with getting the state directly (there are multiple cases in Kiali). The function is called on component render, so there is no problem with automatic refresh.

The redux selector approach that @nrfox mentions would be something like this (not tested):
```
export const isControlPlaneAccessible = (cluster?: string): boolean => {
    return useKialiSelector(state => {
        const ns = state.namespaces;

        return (
          ns.items !== undefined &&
          ns.items.some(nsItem => {
            if (cluster) {
              return nsItem.name === serverConfig.istioNamespace && nsItem.cluster === cluster;
            }

            return nsItem.name === serverConfig.istioNamespace;
          })
        );
      })
```

I don't think so, but if we have any performance issue because of number of namespaces we can use memoized Redux selectors (https://redux.js.org/usage/deriving-data-selectors#writing-memoized-selectors-with-reselect).

Reply:
Thanks for weighing in here @ferhoyos !